crates/ruff_linter/src/rules/pyupgrade/rules/native_literals.rs

```diff
@@ -194,12 +198,16 @@ pub(crate) fn native_literals(
             checker.diagnostics.push(diagnostic);
         }
         Some(arg) => {
+            let Some(literal_expr) = arg.as_literal_expr() else {
+                return;
+            };
+
             // Skip implicit string concatenations.
-            if arg.is_implicit_concatenated_string() {
+            if literal_expr.is_implicit_concatenated() {
```

Comment:
This change is mainly to narrow down the scope for `is_implicit_concatenated` method on literal expressions instead of all expressions (`Expr::is_implicit_concatenated_string`).